### PR TITLE
firefox-mobile: init

### PR DESCRIPTION
**Pull request by bendlas**
###### Description of changes

Adds a ["mobile" configuration for firefox, by postmarketOS](https://gitlab.com/postmarketOS/mobile-config-firefox). This is just policy+css on for the regular firefox (since it now handles wayland natively).

Tested on mobile-nixos on pinephone

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bendlas** [comment]
sorry, totally forgot about this sitting in my patch queue

File: pkgs/applications/networking/browsers/firefox/mobile-config.nix [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        var appStartup = Cc["@mozilla.org/toolkit/app-startup;1"].getService(Ci.nsIAppStartup);
+        appStartup.quit(Ci.nsIAppStartup.eForceQuit | Ci.nsIAppStartup.eRestart);
+    }
+
+    defaultPref('general.useragent.override', 'Mozilla/5.0 (Android 11; Mobile; rv:96.0) Gecko/96.0 Firefox/96.0');
+    defaultPref('browser.urlbar.suggest.topsites', false);
+    defaultPref('browser.urlbar.suggest.engines', false);
+    defaultPref('browser.newtabpage.enabled', true);
+
+    // Enable android-style pinch-to-zoom
```
[the remark this answers is not in the document]
Comment (bendlas) [reply pkgs/applications/networking/browsers/firefox/mobile-config.nix:87]: sorry to not address this before merging, if there is a good way to improve on this, please open a PR

**Comment by uninsane** [comment]
hey, looks like we had two parallel implementations of this package out for PR. i notice this PR's at 3.1.0, whereas upstream is at 4.0.2.

@bendlas if you have a moment, care to take a look over [here](https://github.com/NixOS/nixpkgs/pull/121356#issuecomment-1608619585) to weigh in? i just posted a patch there that gets us inline with upstream, but it's a pretty different approach than what just got merged, so i'm not sure the best way to proceed.

**Comment by SomeoneSerge** [comment]
Awesome! Now  how should this interact with `programs.firefox.package`?

**Pull request merged**